### PR TITLE
[3537][ADD] purchase_invoice_indicator

### PR DESCRIPTION
**Pull request by AungKoKoLin1997**
[3537](https://www.quartile.co/web#view_type=form&model=project.task&id=3537&active_id=3537&menu_id=)

### CODE REVIEW

**Comment by rinaldifirdaus** [comment]
Functional test, works as expected. 
will do translation once vat field added in this module.

File: partner_invoice_target/models/account_invoice.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models,fields
+
+
+class AccountInvoice(models.Model):
+    _inherit = "account.invoice"
+
+    # override vat to add translation
+    vat = fields.Char()
```
Comment (yostashiro) [inline partner_invoice_target/models/account_invoice.py]: What is this for?

File: partner_invoice_target/models/res_partner.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Copyright 2023 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models,fields
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    is_invoice_target = fields.Boolean()
```
Comment (yostashiro) [inline partner_invoice_target/models/res_partner.py]:
```suggestion
    is_invoice_issuer = fields.Boolean("Qualified Invoice Issuer")
```

Assign '適格請求書発行事業者' as the translation.

File: partner_invoice_target/models/account_invoice.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    is_invoice_target = fields.Boolean(related="partner_id.is_invoice_target", store=True, readonly=True)
```
Comment (yostashiro) [inline partner_invoice_target/models/account_invoice.py]:
```suggestion
    is_invoice = fields.Boolean("Is Invoice")
```

We can't make it a related field. @rinaldifirdaus Didn't Kanda-san explain this to you? We should extend the method of purchase.order which prepares values for the invoice.

File: partner_invoice_target/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# Copyright 2023 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+{
+    "name": "Partner Invoice Target",
+    "version": "11.0.1.0.0",
+    "category": "Contact",
+    "website": "https://www.quartile.co",
+    "author": "Quartile Limited",
+    "license": "AGPL-3",
+    "depends": ["account"],
```
Comment (yostashiro) [inline partner_invoice_target/__manifest__.py]:
```suggestion
    "depends": ["purchase_account"],
```

File: partner_invoice_target/views/account_invoice_views.xml
Comment (yostashiro) [inline partner_invoice_target/views/account_invoice_views.xml:1]: Show the field in the form view as well.

File: partner_invoice_target/views/partner_form_views.xml
Comment (yostashiro) [inline partner_invoice_target/views/partner_form_views.xml:1]: Show the field in the list view as well.

**Review by yostashiro — CHANGES_REQUESTED** [review]
Can we rename the module to purchase_invoice_indicator? Please suggest if you have a better idea.

File: partner_invoice_target/models/account_invoice.py (hunk 1 above)
Comment (rinaldifirdaus) [reply partner_invoice_target/models/account_invoice.py]: Hi @AungKoKoLin1997 , the vat field that i ask to be override is the one on res.partner model

File: purchase_invoice_indicator/__manifest__.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    "version": "11.0.1.0.0",
+    "category": "Purchase",
+    "website": "https://www.quartile.co",
+    "author": "Quartile Limited",
+    "license": "AGPL-3",
+    "depends": ["purchase"],
+    "data": [
+                "views/res_partner_views.xml",
+                "views/account_invoice_views.xml"
+            ],
```
Comment (yostashiro) [inline purchase_invoice_indicator/__manifest__.py]: Fix the indentation.

File: purchase_invoice_indicator/i18n/ja.po
Comment (yostashiro) [inline purchase_invoice_indicator/i18n/ja.po:1]: @AungKoKoLin1997 @rinaldifirdaus There is no translation for `vat` - what we should may be to create a PR in oca/ocb to update the original translation to "事業者登録番号" (I'd intentionally make it a bit more generic than "適格請求書発行事業者登録番号").

Comment (AungKoKoLin1997) [reply purchase_invoice_indicator/i18n/ja.po:1]: @rinaldifirdaus  Can you please handle this one in oca/ocb?

File: purchase_invoice_indicator/i18n/ja.po [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+#. module: purchase_invoice_indicator
+#: model:ir.model,name:purchase_invoice_indicator.model_account_invoice
+msgid "Invoice"
+msgstr "請求書"
+
+#. module: purchase_invoice_indicator
+#: model:ir.model.fields,field_description:purchase_invoice_indicator.field_account_invoice_is_invoice
+msgid "Is Invoice"
+msgstr "請求書です"
```
Comment (yostashiro) [inline purchase_invoice_indicator/i18n/ja.po]:
```suggestion
msgstr "適格請求書"
```

**Review by kanda999 — APPROVED** [review]
Function review

**Pull request merged**